index.rst

```diff
@@ -460,7 +441,85 @@ to be accessible from Kafka, and a slow consumer can have additional processing
 can slow the process, but the transit and consumption time should still be considered.  Consumer reading of the
 stream can easily be parallelized for faster processing.  The parallelization is controlled by the number of
 topic partitions, so increasing the number of partitions can decrease consumer read time such that consumers
-do not lag behind bursts if the consumers read in consumer groups in parallel.  For the critical time from
+do not lag behind bursts if the consumers read in consumer groups in parallel, which we explore below.
+For the critical time from
 producer serialization into Avro and submission to Kafka, the time it takes for the alert distribution to get
 alerts into the queue and ready to be read by consumers can be expected to be about 6-9 seconds when stamps
 are included and 1-3 seconds without stamps.
+
+
+Parallelization with Multiple Partitions
+----------------------------------------
+The alert stream we are using is pushed to a single "topic" within Kafka, into which alert messages are
+queued.  Kafka topics can then be divided into a number of "partitions," each of which can be placed on
+separate machines to allow for multiple consumers to read from a topic in parallel.
+We constructed a testbed for multiple partitions using an AWS Docker Swarm cloud of 3 Swarm managers
+on m4.16xlarge instances (64 vCPUs, 256 GiB memory, and 10,000 Mbps EBS bandwidth)
+and 10 worker nodes on m4.4xlarge instances (16 vCPUs, 64 GiB memory, and 2000 Mbps EBS bandwidth) all with
+1 TiB of ephemeral storage, as before.  This simulation used a Kafka cluster of 3 brokers configured to have
+an alert stream topic with 10 partitions, allowing consumers to parallelize into 10 consumer feeds.
+The Kafka containers were constrained to manager nodes and separated from the other alert stream
+producing and consuming containers. We used 100 containers for producers, each producing 100 alerts in
```

Comment:
Looks like you are double spacing after a period but "containers. We" isn't. I'll check for others but I may miss them.

Reply:
fixed.